From fda97bdcd80c5dbced89f64ad340c18987fa39fb Mon Sep 17 00:00:00 2001
From: He Sheng <hesheng@wxiat.com>
Date: Fri, 6 May 2022 14:15:52 +0800
Subject: [PATCH] sw64: remove VGA_HOSE things
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: fda97bdcd80c5dbced89f64ad340c18987fa39fb
Modified-by-SEL: No


Sunway inclusion
category: feature
bugzilla: https://gitee.com/openeuler/kernel/issues/I56OLG

--------------------------------

SW64 has nothing to do with CONFIG_VGA_HOSE.

Signed-off-by: He Sheng <hesheng@wxiat.com>

Signed-off-by: Gu Zitao <guzitao@wxiat.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 arch/sw_64/Kconfig           |  8 --------
 arch/sw_64/include/asm/vga.h | 31 -------------------------------
 2 files changed, 39 deletions(-)

diff --git a/arch/sw_64/Kconfig b/arch/sw_64/Kconfig
index 472a916fd93e..8c5e87cb9b84 100644
--- a/arch/sw_64/Kconfig
+++ b/arch/sw_64/Kconfig
@@ -896,12 +896,4 @@ source "drivers/idle/Kconfig"
 
 endmenu
 
-# DUMMY_CONSOLE may be defined in drivers/video/console/Kconfig
-# but we also need it if VGA_HOSE is set
-config DUMMY_CONSOLE
-	bool
-	depends on VGA_HOSE
-	default y
-
-
 source "arch/sw_64/kvm/Kconfig"
diff --git a/arch/sw_64/include/asm/vga.h b/arch/sw_64/include/asm/vga.h
index 28adb8b8b7f1..7268b5cd4bd8 100644
--- a/arch/sw_64/include/asm/vga.h
+++ b/arch/sw_64/include/asm/vga.h
@@ -49,37 +49,6 @@ extern void scr_memcpyw(u16 *d, const u16 *s, unsigned int count);
 #define vga_readb(a)		readb((u8 __iomem *)(a))
 #define vga_writeb(v, a)	writeb(v, (u8 __iomem *)(a))
 
-#ifdef CONFIG_VGA_HOSE
-#include <linux/ioport.h>
-#include <linux/pci.h>
-
-extern struct pci_controller *pci_vga_hose;
-
-#define __is_port_vga(a) \
-	(((a) >= 0x3b0) && ((a) < 0x3e0) && \
-	((a) != 0x3b3) && ((a) != 0x3d3))
-
-#define __is_mem_vga(a) \
-	(((a) >= 0xa0000) && ((a) <= 0xc0000))
-
-#define FIXUP_IOADDR_VGA(a) do { \
-	if (pci_vga_hose && __is_port_vga(a)) \
-		(a) += pci_vga_hose->io_space->start; \
-} while (0)
-
-#define FIXUP_MEMADDR_VGA(a) do { \
-	if (pci_vga_hose && __is_mem_vga(a)) \
-		(a) += pci_vga_hose->mem_space->start; \
-} while (0)
-
-#else /* CONFIG_VGA_HOSE */
-#define pci_vga_hose 0
-#define __is_port_vga(a) 0
-#define __is_mem_vga(a) 0
-#define FIXUP_IOADDR_VGA(a)
-#define FIXUP_MEMADDR_VGA(a)
-#endif /* CONFIG_VGA_HOSE */
-
 #define VGA_MAP_MEM(x, s)	((unsigned long)ioremap(x, s))
 
 #endif
-- 
2.33.0

